### PR TITLE
Deprecate v1alpha1 and v1beta1 models, correct API version, & hiberante dashboard

### PR DESCRIPTION
**Pull request by frrist**
**Note: this change targets `dev1`; subsequent breaking changes will continue to target `dev1`.**
### What
- Remove all references to `v1alpha1` and `v1beta1` job specifications.
- Assign a new (and correct) version to the current model specification `v1beta2`.
- Hibernate the dashboard (due to breakages from `v1alpha1` and `v1beta1` removal)
  - https://github.com/bacalhau-project/bacalhau/issues/2536 tracks re-adding support for dashboard

Closes https://github.com/bacalhau-project/bacalhau/issues/2535 & #2551

### CODE REVIEW

**Comment by rossjones** [comment]
-18,683 ... nice. I love a good tidy up. But I have concerns around hibernating the dashboard on a branch that is intended to be long lived.  Unless we've already agreed this with product peeps, we're essentially committing to refactoring the dashboard into a workable state before we ever merge in dev1. Is that intentional?

File: pkg/model/apiversion.go [hunk 1]
```diff
@@ -9,8 +9,7 @@ type APIVersion int
 
 const (
 	apiVersionUnknown APIVersion = iota // must be first
-	V1alpha1
-	V1beta1
+	V0
```
Comment (frrist) [inline pkg/model/apiversion.go:12]: @wdbaruni Are we okay calling this version of the spec `V0`?

Comment (wdbaruni) [reply pkg/model/apiversion.go:12]: better to use `V1gamma`. V0 indicates it is the very first version but in this case it isn't

**Comment by frrist** [comment]
@rossjones inlight of @wdbaruni's design on a module API and plans for backwards compatibility this will now go into `main`

**Comment by frrist** [comment]
For some reason, CI is failing to install the precommit hook; once I get this sorted, I'll get this merged.

**Comment by frrist** [comment]
@rossjones and @simonwo this is a relatively large change, so when you have time I'd like to get a review and ✅ from both of you before merging.

**Review by rossjones — APPROVED** [review]
I'm not sure where we got to with the discussion about insulated jobs, but removing the old models looks good!

**Pull request merged**